don't just rely on ldflags to generate version

chmouel opened the pull request:
set it on release or nightly directly, so docker/cpaas and other who
don't use the make target or goreleaser works

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [ ] ♽  Run `make test lint` before submitting a PR (ie: with [pre-commit](https://pipelinesascode.com/dev/tools), no need to waste CPU cycle on CI
- [ ] 📖 If you are adding a user facing feature or make a change of the behavior, please verify that you have documented it
- [ ] 🧪 100% coverage is not a target but most of the time we would rather have a unit test if you make a code change.
- [ ] 🎁 If that's something that is possible to do please ensure to check if we can add a e2e test.
- [ ] 🔎 If there is a flakiness in the CI tests then don't *necessary* ignore it, better get the flakyness fixed before merging or if that's not possible there is a good reason to bypass it. (token rate limitation may be a good reason to skip).

chmouel committed:
don't just rely on ldflags to generate version

set it on release or nightly directly, so docker/cpaas and other who
don't use the make target or goreleaser works

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

piyush-garg commented on .tekton/release-pipeline.yaml:
```diff
@@ -77,8 +77,8 @@ spec:
                   set +x
-                  echo python hack/upload-file-to-github.py --message "Release yaml generated for ${msg}" --owner-repository openshift-pipelines/pipelines-as-code --token "TOKEN" --to-ref=refs/heads/${target} --from-ref=refs/tags/${version} -f release.k8s.yaml:release.k8s.yaml -f release.yaml:release.yaml -f ${TARGET_BRANCH}:docs/content/VERSION
-                  python hack/upload-file-to-github.py --message "Release yaml generated for ${msg}" --owner-repository openshift-pipelines/pipelines-as-code --token ${HUB_TOKEN} --to-ref=refs/heads/${target} --from-ref=refs/tags/${version} -f /tmp/release.k8s.yaml:release.k8s.yaml -f /tmp/release.yaml:release.yaml -f ${TARGET_BRANCH}:docs/content/VERSION
+                  echo python hack/upload-file-to-github.py --message "Release yaml generated for ${msg}" --owner-repository openshift-pipelines/pipelines-as-code --token "TOKEN" --to-ref=refs/heads/${target} --from-ref=refs/tags/${version} -f release.k8s.yaml:release.k8s.yaml -f release.yaml:release.yaml -f ${TARGET_BRANCH}:docs/content/VERSION -f "${TARGET_BRANCH}:pkg/params/version/version.txt ...
```
we need to put the version in `version.txt`

piyush-garg replied:
ohh I see TARGET_BRANCH contains the version from the release branch after removing the `release-`, the name was confusing

chmouel replied:
yeah it's badly named

chmouel committed:
Remove new line from kubernetes labels

or it will fail like this:

```
Tekton Controller has reported this error: PipelineRun.tekton.dev "pipelinerun-msco-gr4kd" is invalid: metadata.labels: Invalid value: "nightly\n": a va>
```

do the same everywhere we use version.Version, we could fix the
version.txt as well but it's  a pain to make sure we never going to
add it back.

Fixes #1271

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>
